server.js

```diff
@@ -24,9 +29,113 @@ const app = express();
 app.use(cors());
 app.use(express.json());
 
-// Start defining your routes here
-app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
+const Song = mongoose.model('Song', {
+  id: Number,
+  trackName: String,
+  artistName: String,
+  genre: String,
+  bpm: Number,
+  energy: Number,
+  danceability: Number,
+  loudness: Number,
+  liveness: Number,
+  valence: Number,
+  length: Number,
+  acousticness: Number,
+  speechiness: Number,
+  popularity: Number,
+  danceability: Number
+});
+// populate database with music data
+if (process.env.RESET_DB) {
+  console.log('Resetting database!');
+  const seedDB = async () => {
+    await Song.deleteMany();
+
+    musicData.forEach((songData) => {
+      const song = new Song(songData);
+      console.log(song);
+      song.save();
+    });
+  };
+  seedDB();
+}
+// Check that the database is running
+app.use((req, res, next) => {
+  if (mongoose.connection.readyState === 1) {
+    next();
+  } else {
+    res.status(503).json({ error: 'Service unavailable' });
+  }
+});
+
+// Root endpoint (with list of endpoints)
+app.get('/', (req, res) => {
+  res.json(listEndpoints(app));
+});
+
+// Endpoint to get all songs
+app.get('/songs', async (req, res) => {
+  // can only await something in an async
+  try {
+    const songs = await Song.find();
+    res.status(200).json({
+      success: true,
+      body: songs
+    });
+  } catch (error) {
+    res.status(500).json({
+      success: false,
+      body: {
+        message: error
+      }
+    });
+  }
+});
+
+// Endpoint to get a specific song
+app.get('/songs/:id', (req, res) => {
+  const { id } = req.params;
+  const singletrack = musicData.find((track) => {
+    return track.id === Number(id);
+  });
+  if (singletrack) {
+    res.status(200).json({
+      success: true,
+      message: 'OK',
+      body: {
+        track: singletrack
+      }
+    });
+  } else {
+    res.status(404).json({
+      success: false,
+      message: 'Track not found',
+      body: {}
+    });
+  }
+});
+
+//songs by bpm
+app.get('/songs/bpm/:bpm', async (req, res) => {
+  const { bpm } = req.params;
+  const songs = await Song.find({ bpm: bpm });
+  if (songs.length > 0) {
+    res.json(songs);
+  } else {
+    res.status(404).json({ error: 'BPM not found' });
+  }
+});
+
+// endpoint to get songs by genre
+app.get('/genres/:genre', async (req, res) => {
+  const { genre } = req.params;
+  const songs = await Song.find({ genre: genre });
```

Comment:
Daniel showed us a way to improve this search, as it is now when you search for example pop you will only receive the exact matching category. Since there are many different pop categories, maybe you would like to look into this solution:
Create a case insensitive regex:
`const genreRegex = new RegExp(genre, 'i');`
When using this it will include more matches:
`const songs = await Song.find({ genre: genreRegex })`